BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -55,7 +55,11 @@ public class NetworkStore: ObservableObject {
       // we don't need to call `setNetwork` on JsonRpcService
       self.selectedChainId = chain.chainId
       // update `isSwapSupported` for Buy/Send/Swap panel
-      self.isSwapSupported = await swapService.isSwapSupported(chain.chainId)
+      if chain.coin == .eth {
```

Comment:
I think we need this check in `SwapTokenStore` too for when `isSwapSupported` returns true for Solana in v1.43 of Brave Core?

Reply:
inside `SwapTokenStore` is using `NetworkStore.isSwapSupported` which is always updated with the logic of checking network coin type.